compactor/periodic.go

```diff
@@ -87,26 +86,36 @@ func (t *Periodic) Run() {
 			if t.clock.Now().Sub(initialWait) < t.period {
 				continue
 			}
-
-			rev, remaining := t.getRev()
-			if rev < 0 {
-				continue
-			}
+			rev := t.revs[0]
 
 			plog.Noticef("Starting auto-compaction at revision %d (retention: %v)", rev, t.period)
 			_, err := t.c.Compact(t.ctx, &pb.CompactionRequest{Revision: rev})
 			if err == nil || err == mvcc.ErrCompacted {
 				// move to next sliding window
-				t.revs = remaining
+				t.revs = t.revs[1:]
 				plog.Noticef("Finished auto-compaction at revision %d", rev)
 			} else {
+				// retry fast
+				interval = t.getInterval() / 10
```

Comment:
I think in retry iterations, we don't want to update `t.revs`.
Maybe something like below is required at line 70.

```go
if interval == t.getInterval() {
  t.revs = append(t.revs, t.rg.Rev())
}
```

Reply:
@yudai Good catch. Just fixed. PTAL. Thanks.

Reply:
Ah sorry, I have another corner case.

Now `t.revs` is not updated while retry is ongoing. But if compaction keeps failing for a while, it will be outdated. So ideally, we need to update `t.revs` and shift the window once every 10 times when retry is ongoing to keep `t.revs[0]` fresh.

Reply:
One more thing. I think it's just about definition of "interval". If retry fails 9 times, next compaction with a new rev happens at `(9/10 + 1) hour`. So the effective interval and the DB overhead can vary between 1 hour through 1.9 hours.

I guess without using two goroutines, these are hard to implement simpler. You may want to put these corner cases aside in this PR.